node-registrar/pkg/server/handlers.go

```diff
@@ -281,6 +281,69 @@ func (s Server) getNodeHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, node)
 }
 
+// @Summary Get node monthly reward information
+// @Description Retrieves reward calculation for a specific node based on resources and uptime
+// @Tags nodes
+// @Accept json
+// @Produce json
+// @Param node_id path int true "Node ID"
+// @Success 200 {object} Reward "Rewards details with the node uptime percentage"
+// @Failure 400 {object} map[string]any "Invalid node ID"
+// @Failure 404 {object} map[string]any "Node not found"
+// @Failure 422 {object} map[string]any "No uptime reports available for this node"
+// @Router /nodes/{node_id}/rewards [get]
+func (s Server) getNodeRewardHandler(c *gin.Context) {
+	nodeID := c.Param("node_id")
+
+	id, err := strconv.ParseUint(nodeID, 10, 64)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid node id"})
+		return
+	}
+
+	node, err := s.db.GetNode(id)
+	if err != nil {
+		if errors.Is(err, db.ErrRecordNotFound) {
+			c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+			return
+		}
+
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	now := time.Now()
+	periodStart := calculatePeriodStart(now)
+
+	reports, err := s.db.GetUptimeReports(id, periodStart, now)
+
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	upTimePercentage, err := calculateUpTimePercentage(reports, periodStart, now)
+	if err != nil {
+		if err == ErrReportsNotInAscOrder {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
+		if err == ErrNoReportsAvailable {
+			c.JSON(http.StatusUnprocessableEntity, gin.H{"error": err.Error()})
+			return
+		}
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
```

Comment:
if you return status internal server error in case of `ErrReportNotInAscOrder` and in the default case then maybe you should just remove the check for `ErrReportNotInAscOrder` and only use the default
```suggestion
		if errors.Is(err, ErrNoReportsAvailable) {
			c.JSON(http.StatusUnprocessableEntity, gin.H{"error": err.Error()})
			return
		}
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return

```

Reply:
I think we are passing the error, so it may help us in debugging.

Reply:
we're passing the error in both cases

Reply:
got it thank you